### PR TITLE
[Workers] add upstash integration

### PR DESCRIPTION
**Pull request by tanushree-sharma**

### CODE REVIEW

File: content/workers/databases/native-integrations/upstash.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
+---
+pcx_content_type: concept
+title: Upstash
+---
+## Connecting to Upstash
```
Comment (deadlypants1973) [inline content/workers/databases/native-integrations/upstash.md]:
```suggestion
# Upstash
```

File: content/workers/databases/native-integrations/upstash.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+[Upstash](https://upstash.com/) is a serverless database with Redis and Kafka API. Upstash also offers QStash, a task queue/scheduler designed for the serverless. 
+
```
Comment (deadlypants1973) [inline content/workers/databases/native-integrations/upstash.md:8]:
```suggestion
```

File: content/workers/databases/native-integrations/upstash.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+### Set up an integration with Upstash Redis:
```
Comment (deadlypants1973) [inline content/workers/databases/native-integrations/upstash.md]:
```suggestion
## Upstash Redis
```

File: content/workers/databases/native-integrations/upstash.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+1. You need an existing Upstash database to connect to. [Create a Upstash database](https://docs.upstash.com/redis#create-a-database) or [load data from an existing database to Upstash](https://docs.upstash.com/redis/howto/connectclient).
```
Comment (deadlypants1973) [inline content/workers/databases/native-integrations/upstash.md:13]:
```suggestion
To set up an integration with Upstash:

1. You need an existing Upstash database to connect to. [Create a Upstash database](https://docs.upstash.com/redis#create-a-database) or [load data from an existing database to Upstash](https://docs.upstash.com/redis/howto/connectclient).
```

File: content/workers/databases/native-integrations/upstash.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+2. Insert some data to your Upstash database. You can add data to your Upstash database in two ways:
```
Comment (deadlypants1973) [inline content/workers/databases/native-integrations/upstash.md:15]:
```suggestion

2. Insert some data to your Upstash database. You can add data to your Upstash database in two ways:
```

File: content/workers/databases/native-integrations/upstash.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+3. Add the Upstash Redis integration to your Worker:
+    - Log in to the [Cloudflare dashboard](https://dash.cloudflare.com) and select your account.
+    - In **Account Home**, select **Workers & Pages**.
+    - In **Overview**, select your Worker.
+    - Select **Settings** > **Integrations** > **Upstash Redis**. 
+    - Follow the setup flow, selecting the database created in step 1.
+4. In your Worker, install the `@upstash/redis`, a HTTP client to connect to your database and start manipulating data:
+    ```sh
+    $ npm install @upstash/redis
+    ```
```
Comment (deadlypants1973) [inline content/workers/databases/native-integrations/upstash.md]:
```suggestion
3. Add the Upstash Redis integration to your Worker:

    a. Log in to the [Cloudflare dashboard](https://dash.cloudflare.com) and select your account.
    b. In **Account Home**, select **Workers & Pages**.
    c. In **Overview**, select your Worker.
    d. Select **Settings** > **Integrations** > **Upstash Redis**. 
    e. Follow the setup flow, selecting the database created in step 1.
    
4. In your Worker, install the `@upstash/redis`, a HTTP client to connect to your database and start manipulating data:

    ```sh
    $ npm install @upstash/redis
    ```

```

File: content/workers/databases/native-integrations/upstash.md [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+5. The following example shows how to make a query to your Upstash database in a Worker. The credentials needed to connect to Upstash have been automatically added as secrets to your Worker through the integration.
+  ``` js
+  import { Redis } from "@upstash/redis/cloudflare";
+
+  export default {
+    async fetch(request, env) {
+      const redis = Redis.fromEnv(env);
+
```
Comment (deadlypants1973) [inline content/workers/databases/native-integrations/upstash.md]:
```suggestion
5. The following example shows how to make a query to your Upstash database in a Worker. The credentials needed to connect to Upstash have been automatically added as secrets to your Worker through the integration.

  ``` js
  ---
  filename: Worker code
  ---
  import { Redis } from "@upstash/redis/cloudflare";

  export default {
    async fetch(request, env) {
      const redis = Redis.fromEnv(env);

```

File: content/workers/databases/native-integrations/upstash.md [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+      return new Response("Hello What's up!");
+    },
+  };
+  ```
+
+To learn more about Upstash, refer to the [Upstash documentation](https://docs.upstash.com/redis).
+
+### Set up an integration with Upstash Kafka:
+1. Create a [Kafka cluster and topic](https://docs.upstash.com/kafka)
+2. Add the Upstash Kafka integration to your Worker:
```
Comment (deadlypants1973) [inline content/workers/databases/native-integrations/upstash.md]:
```suggestion
## Upstash Kafka

To set up an integration with Upstash Kafka:

1. Create a [Kafka cluster and topic](https://docs.upstash.com/kafka).

2. Add the Upstash Kafka integration to your Worker:

```

File: content/workers/databases/native-integrations/upstash.md [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+    - Log in to the [Cloudflare dashboard](https://dash.cloudflare.com) and select your account.
+    - In **Account Home**, select **Workers & Pages**.
+    - In **Overview**, select your Worker.
+    - Select **Settings** > **Integrations** > **Upstash Kafka**. 
+    - Follow the setup flow, selecting the cluster and topic.
+3. In your Worker, install `@upstash/kafka`, a HTTP/REST based Kafka client: 
+    ```sh
+    $ npm install @upstash/kafka
+    ```
```
Comment (deadlypants1973) [inline content/workers/databases/native-integrations/upstash.md]:
```suggestion
    a. Log in to the [Cloudflare dashboard](https://dash.cloudflare.com) and select your account.
    b. In **Account Home**, select **Workers & Pages**.
    c. In **Overview**, select your Worker.
    d. Select **Settings** > **Integrations** > **Upstash Kafka**. 
    e. Follow the setup flow, selecting the cluster and topic.
    
3. In your Worker, install `@upstash/kafka`, a HTTP/REST based Kafka client:

    ```sh
    $ npm install @upstash/kafka
    ```

```

File: content/workers/databases/native-integrations/upstash.md [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+4. Use the [upstash-kafka](https://github.com/upstash/upstash-kafka/blob/main/README.md) JavaScript SDK to send data to Kafka.
+Check out an [example](https://docs.upstash.com/kafka/real-time-analytics/realtime_analytics_serverless_kafka_setup#option-1-cloudflare-workers) published by Upstash. Replace `url`, `username` and `password` with the variables set by the integration.
```
Comment (deadlypants1973) [inline content/workers/databases/native-integrations/upstash.md]:
```suggestion
4. Use the [upstash-kafka](https://github.com/upstash/upstash-kafka/blob/main/README.md) JavaScript SDK to send data to Kafka.

Refer to [Upstash documentation on Kafka setup with Workers](https://docs.upstash.com/kafka/real-time-analytics/realtime_analytics_serverless_kafka_setup#option-1-cloudflare-workers) for more information. Replace `url`, `username` and `password` with the variables set by the integration. 
```

File: content/workers/databases/native-integrations/upstash.md [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+    - Follow the setup flow, selecting the cluster and topic.
+3. In your Worker, install `@upstash/kafka`, a HTTP/REST based Kafka client: 
+    ```sh
+    $ npm install @upstash/kafka
+    ```
+4. Use the [upstash-kafka](https://github.com/upstash/upstash-kafka/blob/main/README.md) JavaScript SDK to send data to Kafka.
+Check out an [example](https://docs.upstash.com/kafka/real-time-analytics/realtime_analytics_serverless_kafka_setup#option-1-cloudflare-workers) published by Upstash. Replace `url`, `username` and `password` with the variables set by the integration. 
+
+### Set up an integration with Upstash QStash:
+1. Configure the [publicly available HTTP endpoint](https://docs.upstash.com/qstash#1-public-api) that you want to send your messages to.
```
Comment (deadlypants1973) [inline content/workers/databases/native-integrations/upstash.md]:
```suggestion
## Upstash QStash

To set up an integration with Upstash QStash:

1. Configure the [publicly available HTTP endpoint](https://docs.upstash.com/qstash#1-public-api) that you want to send your messages to.

```

File: content/workers/databases/native-integrations/upstash.md [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+Check out an [example](https://docs.upstash.com/kafka/real-time-analytics/realtime_analytics_serverless_kafka_setup#option-1-cloudflare-workers) published by Upstash. Replace `url`, `username` and `password` with the variables set by the integration. 
+
+### Set up an integration with Upstash QStash:
+1. Configure the [publicly available HTTP endpoint](https://docs.upstash.com/qstash#1-public-api) that you want to send your messages to. 
+2. Add the Upstash QStash integration to your Worker:
+    - Log in to the [Cloudflare dashboard](https://dash.cloudflare.com) and select your account.
+    - In **Account Home**, select **Workers & Pages**.
+    - In **Overview**, select your Worker.
+    - Select **Settings** > **Integrations** > **Upstash QStash**. 
+    - Follow the setup flow..
```
Comment (deadlypants1973) [inline content/workers/databases/native-integrations/upstash.md]:
```suggestion
2. Add the Upstash QStash integration to your Worker:

    a. Log in to the [Cloudflare dashboard](https://dash.cloudflare.com) and select your account.
    b. In **Account Home**, select **Workers & Pages**.
    c. In **Overview**, select your Worker.
    d. Select **Settings** > **Integrations** > **Upstash QStash**. 
    e. Follow the setup flow.
    
```
Was there supposed to be something after follow the setup flow?

File: content/workers/databases/native-integrations/upstash.md [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+3. In your Worker, install the `@upstash/qstash`, a HTTP client to connect to your database QStash endpoint:
+    ```sh
+    $ npm install @upstash/qstash
+    ```
+4. Check out an [example](https://docs.upstash.com/qstash/quickstarts/cloudflare-workers#3-use-qstash-in-your-handler) published by Upstash on how to receive webhooks from QStash in your Cloudflare Worker.
```
Comment (deadlypants1973) [inline content/workers/databases/native-integrations/upstash.md]:
```suggestion
3. In your Worker, install the `@upstash/qstash`, a HTTP client to connect to your database QStash endpoint:
    ```sh
    $ npm install @upstash/qstash
    ```
4. Refer to the [Upstash documentation on how to receive webhooks from QStash in your Cloudflare Worker](https://docs.upstash.com/qstash/quickstarts/cloudflare-workers#3-use-qstash-in-your-handler). 
```

**Review by deadlypants1973 — APPROVED** [review]
PCX

File: content/workers/databases/native-integrations/upstash.md (hunk 12 above)
Comment (tanushree-sharma) [reply content/workers/databases/native-integrations/upstash.md]: Oops typo. Thats the last step.

**Pull request merged**